src/services/helpers/workspaceAdapter.ts

```diff
@@ -0,0 +1,229 @@
+    }
+  }
+
+  get infrastructureNamespace(): string {
+    if (isWorkspaceV1(this.workspace)) {
+      return this.workspace.attributes?.infrastructureNamespace || '';
+    } else {
+      return (this.workspace as IDevWorkspace).metadata.namespace;
+    }
+  }
+
+  get created(): number {
+    if (isWorkspaceV1(this.workspace)) {
+      return parseInt(this.workspace.attributes?.created || '', 10) || 0;
+    } else {
+      const reference = this.workspace as IDevWorkspace;
+      if (!reference.metadata.creationTimestamp) {
+        return 0;
+      }
+      return new Date(reference.metadata.creationTimestamp).valueOf();
+    }
+  }
+
+  get updated(): number {
+    if (isWorkspaceV1(this.workspace)) {
+      return parseInt(this.workspace.attributes?.updated || '', 10) || 0;
+    } else {
+      const reference = this.workspace as IDevWorkspace;
+      if (!reference.metadata.creationTimestamp) {
+        return 0;
+      }
+      return new Date(reference.metadata.creationTimestamp).valueOf();
+    }
+  }
+
+  get status(): keyof typeof WorkspaceStatus {
+    if (isWorkspaceV1(this.workspace)) {
+      return this.workspace.status as keyof typeof WorkspaceStatus;
+    } else {
+      let status = (this.workspace as IDevWorkspace).status?.phase;
+      if (status) {
+        status = status.toUpperCase();
+        if (status === DevWorkspaceStatus[DevWorkspaceStatus.FAILED]) {
+          status = WorkspaceStatus[WorkspaceStatus.ERROR];
+        }
+      }
+      return (status || '') as keyof typeof WorkspaceStatus;
+    }
+  }
+
+  set status(status: keyof typeof WorkspaceStatus) {
+    if (isWorkspaceV1(this.workspace)) {
+      this.workspace.status = status;
+    } else {
+      (this.workspace as IDevWorkspace).status.phase = status;
+    }
+  }
+
+  get ideUrl(): string | undefined {
+    if (isWorkspaceV1(this.workspace)) {
+      const runtime = this.workspace.runtime;
+      if (!runtime || !runtime.machines) {
+        return;
+      }
+      for (const machineName of Object.keys(runtime.machines)) {
+        const servers = runtime.machines[machineName].servers || {};
+        for (const serverId of Object.keys(servers)) {
+          const attributes = (servers[serverId] as any).attributes;
+          if (attributes && attributes['type'] === 'ide') {
+            return servers[serverId].url;
+          }
+        }
+      }
+    } else {
+      return (this.workspace as IDevWorkspace).status.ideUrl;
+    }
+  }
+
+  get storageType(): che.WorkspaceStorageType {
+    if (isWorkspaceV1(this.workspace)) {
+      return attributesToType(this.workspace.devfile.attributes);
+    } else {
+      throw new Error('not implemented.');
+    }
+  }
+
+  set storageType(type: che.WorkspaceStorageType) {
+    if (isWorkspaceV1(this.workspace)) {
+      const attributes = typeToAttributes(type);
+      Object.assign(this.workspace.devfile.attributes, attributes);
+    } else {
+      throw new Error('not implemented.');
```

Comment:
DevWorkspace details does not work, probably due this error.

![Screenshot_20210408_103012](https://user-images.githubusercontent.com/5887312/113986063-7eaf6c80-9855-11eb-8ce5-df1e0644b622.png)

Reply:
should be fixed